### DISCUSSION TITLE
Hyperlink not detected

### DISCUSSION THREAD

**Discussion by tcn33** [ghostty-org/ghostty#3635] [Q&A] [answered]
When I run `npm run dev` Ghostty is not parsing `http://localhost:4321` in the output as a hyperlink. (It's not just that it's not underlined, clicking on it does nothing.)

<img width="753" alt="SCR-20241228-mevz" src="https://github.com/user-attachments/assets/f59cf3da-2ff2-4340-b752-e6b2b4e021e0" />

**Comment by mitchellh** [comment, marked as answer]
Hold cmd or ctrl.

**Reply by tcn33** [reply]
Aha! Thank you - I am coming from Warp where the link and cursor change on mouseover, and a `Open link [Cmd + click]` tooltip explains how to use it.
<img width="454" alt="SCR-20241230-mywz" src="https://github.com/user-attachments/assets/5f7ee339-9dc2-4ca6-b57e-930aded7ae4d" />

**Reply by BearyDevs** [reply]
Cmd + click is not work for my Ghostty. How can I get it?

**Reply by 00-kat** [reply]
@BearyDevs if you're in a fullscreen app you'll need to use cmd+shift+click.

**Reply by pluiedev** [reply]
> and a `Open link [Cmd + click]` tooltip explains how to use it. 

That could be a potential improvement I think, UI/UX-wise. Feel free to open a new discussion if you'd like to pursue this further.

**Reply by paulgessinger** [reply]
Hi! This does not seem to work inside a `tmux` session. Any idea how that would be different?

**Reply by 00-kat** [reply]
@paulgessinger https://github.com/ghostty-org/ghostty/discussions/3635#discussioncomment-12310395

**Comment by yujunz** [comment]
Can we customize the cursor to 👆 when holding cmd over links?

**Reply by pluiedev** [reply]
![image](https://github.com/user-attachments/assets/b9a1af89-19aa-43d2-86ea-38f27ca23476)
We already do this on Linux at least

**Reply by 00-kat** [reply]
There seem to be lots of issues with cursor shape on macOS open, see #5384 and #3565.  Perhaps a more general discussion that encompasses all three of these (the two aforementioned discussions, as well as a pointing cursor for links) should be opened.

**Reply by bretonics** [reply]
Using macOS and missed it too, thinking it was not a feature or it was disabled.

Prior experience has set expectations that it at least highlights on cursor hover, e.g. VS Code.

This would be nice to have.

**Comment by laixintao** [comment]
Hi seems can not detect links inside tmux in side ghostty.

**Reply by 00-kat** [reply]
Hold <kbd>ctrl</kbd>/<kbd>cmd</kbd>+<kbd>shift</kbd>.

**Reply by laixintao** [reply]
cmd+shift works thanks so much!

previously it's cmd + click now added shift? didn't notice this change!

**Reply by laixintao** [reply]
btw, <kbd>ctrl</kbd> not working

**Reply by 00-kat** [reply]
> previously it's cmd + click now added shift? didn't notice this change!

It never changed; if the application requests mouse events (i.e. `set -g mouse on` in your Tmux config), it's <kbd>cmd</kbd>+<kbd>shift</kbd>+click, otherwise it's just plain <kbd>cmd</kbd>+click.

> btw, <kbd>ctrl</kbd> not working

Yes, usually people write <kbd>ctrl</kbd>/<kbd>cmd</kbd> when they mean to say “<kbd>cmd</kbd> on macos, <kbd>ctrl</kbd> elsewhere”, not “both <kbd>cmd</kbd> and <kbd>ctrl</kbd> on macOS”.
